slam-ws/src/load_waypoints/scripts/load_waypoints_server.py

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env python
+
+import rospy, os, json 
+from load_waypoints.srv import *
+
+all_waypoints = list()
+
+def populate_waypoint_table():
+    base_dir = rospy.get_param('~arg_name') # ~ added to arg_name because private param 
+    
+    with open(base_dir + '/scripts/waypoints.json') as f:
+        try:
+            data = json.load(f)
+        except ValueError, e:
+            rospy.loginfo("Invalid JSON")
+            sys.exit(1)
+
+    # Parse through dictionary and create list of lists holding all waypoints
+    for waypoint in data["waypoints"]:
+        all_waypoints.append([waypoint['coordinate 1'], waypoint['coordinate 2']])
+
+    rospy.loginfo("Loaded waypoints: %s", all_waypoints)
+
+    return 
+
+def handle_waypoint_request(waypoint_request):
+    rospy.loginfo("Returning request for waypoint #%s "%(waypoint_request.waypoint_number))
+    return WaypointRequestResponse(all_waypoints[waypoint_request.waypoint_number])
```

Comment:
Also forgot about this earlier...

We should validate the request here since it's possible that we request a waypoint number that doesn't exist.  We should ensure that `0 <= waypoint_request.waypoint_number <= len(all_waypoints )`.

There are two ways we can handle this:

1. Throw an error.  We'll need to make sure the node actually keeps running and any calling node will need to implement exception handling.

2. Modify our response type to also include a valid flag, and a description. Then, if the request is invalid, we could set this flag to `false`, and provide a description for why.

What do you think?

Reply:
2. is probably better, so that we can ensure that the service node keeps running.

Reply:
Let's do it!